clusterloader2/pkg/test/context.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+
+	"k8s.io/perf-tests/clusterloader2/pkg/framework"
+	"k8s.io/perf-tests/clusterloader2/pkg/state"
+)
+
+// Context is a global test context.
+type Context interface {
+	GetFramework() *framework.Framework
+	GetPath() string
+	Step(i int) StepContext
+}
+
+// StepContext is a step level context.
+type StepContext interface {
+	GetPath() string
```

Comment:
Why do you need path for step?

Reply:
See my response to https://github.com/kubernetes/perf-tests/pull/157#discussion_r202322149. You shouldn't pollute an interface with implementation-specific helpers, etc..